### PR TITLE
Refactor & Cleanup

### PR DESCRIPTION
**Pull request by JackTheFoxOtter**

### CODE REVIEW

File: docker-compose.dev.yml
Comment (LsHallo) [inline docker-compose.dev.yml:1]: you can change this for your local dev setup but please don't commit the changes

File: src/certbot.py [hunk 1]
```diff
@@ -1,81 +1,95 @@
-import os
+# Custom logging setup
+from modules.logging import setup_logging, get_logger
+setup_logging()
+
+# Parse & get environment variables
+from modules.env import *
+
+# Create application logger
+logger = get_logger('Certbot Multidomain', IS_DEBUG)
```
Comment (LsHallo) [inline src/certbot.py]: code above imports?
not a fan

File: src/certbot.py [hunk 2]
```diff
@@ -1,81 +1,95 @@
...
-    global config
-    logging.debug(f"Opening config file f{path}")
-    with open(path, 'r') as f:
-        try:
-            config = yaml.safe_load(f)
-        except yaml.YAMLError as e:
-            logging.error(repr(e))
+logger.debug(f"Opening & parsing config file '{CONFIG_PATH}'...")
+with open(CONFIG_PATH, 'r') as f:
+    config = yaml.safe_load(f)
```
Comment (LsHallo) [inline src/certbot.py:27]: not a fan of the removed try except block

File: src/certbot.py [hunk 3]
```diff
@@ -1,81 +1,95 @@
...
-    logging.info(stdout)
-    logging.error(stderr)
+    logger.info(stdout)
+    logger.error(stderr)
 
     # Check for valid config and reload nginx
-    logging.debug("Reloading nginx container configuration to apply certs.")
-    ret_code = os.system(f"docker exec {os.getenv('NGINX_CONTAINER_NAME', 'nginx')} sh -c 'nginx -t && nginx -s reload'")
+    logger.debug(f"Reloading Nginx configuration for container '{NGINX_CONTAINER_NAME}' to apply certs.")
+    ret_code = os.system(f"docker exec {NGINX_CONTAINER_NAME} sh -c 'nginx -t && nginx -s reload'")
```
Comment (LsHallo) [inline src/certbot.py:76]: maybe also use subprocess

File: src/certbot.py (hunk 1 above)
Comment (JackTheFoxOtter) [reply src/certbot.py]: Fine, I'll move it down.

File: src/certbot.py (hunk 3 above)
Comment (LsHallo) [reply src/certbot.py:76]: will do that later. resolving here

**Review by LsHallo — APPROVED** [review]

**Review by LsHallo — APPROVED** [review]

**Pull request merged**